### PR TITLE
TFAC-1114 Create new UserEventLog Table

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

File: graphql/database/logs/logTypes.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import types from '../fieldTypes'
+
```
Comment (kmjennison) [inline graphql/database/logs/logTypes.js:2]: The typing here isn't strictly necessary but is probably helpful as a guide to others consuming data and as one additional check on data soundness. I'm on board with it.

File: graphql/database/logs/UserEventLogModel.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import tableNames from '../tables'
+import { permissionAuthorizers } from '../../utils/authorization-helpers'
+import { VALID_LOG_TYPES } from './logTypes'
+
+/*
+ * @extends DynamoDBModel
+ */
+class UserEventLogModel extends DynamoDBModel {
+  static get name() {
+    return ''
```
Comment (kmjennison) [inline graphql/database/logs/UserEventLogModel.js:12]: Will need a table name here.

File: graphql/database/logs/logTypes.js (hunk 1 above)
Comment (jedtan) [reply graphql/database/logs/logTypes.js:2]: Yeah I think the former was from my perspective as it serves as documentation in one spot for the differing log schemas.

File: graphql/database/logs/UserEventLogModel.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+/*
+ * @extends DynamoDBModel
+ */
+class UserEventLogModel extends DynamoDBModel {
+  static get name() {
+    return USER_EVENT_LOG
+  }
+
+  static get hashKey() {
+    return 'userId'
```
Comment (jedtan) [inline graphql/database/logs/UserEventLogModel.js:17]: @kmjennison Would like your thoughts on this. If we happen to want to write two logs in the same mutation technically the hashKey/rangeKey could collide.

Comment (jedtan) [reply graphql/database/logs/UserEventLogModel.js:17]: I feel like something ideal would be event type / user id / timestamp; is there a way to express this that way?

Comment (kmjennison) [reply graphql/database/logs/UserEventLogModel.js:17]: That said, I've elsewhere cut corners on the userId+timestamp index problem by assuming some small number of item creations will conflict and just making micro-adjustments to the timestamps:
https://github.com/gladly-team/tab/blob/master/graphql/database/users/logTab.js#L59-L63

**Review by kmjennison — COMMENTED** [review]
Awaiting your opinion/edits on table design.

File: graphql/database/logs/UserEventLogModel.js (hunk 3 above)
Comment (jedtan) [reply graphql/database/logs/UserEventLogModel.js]: That's definitely a creative way to do it!

**Review by kmjennison — APPROVED** [review]
LGTM! ☕

**Pull request merged**